File: client/src/app/_models/order.ts
import { Address } from "./address";
import { RealProduct } from "./realProduct";
import { User } from "./user";

export interface Order {
    id?: number;
    orderDate?: Date;

    requiredDate: Date;
    shippedDate?: Date;
    
    requiredReturnDate: Date;
    returnDate?: Date;

    status?: string;
    comments?: string;

    
    shippedAddressId: number;
    shippedAddress?: Address;
    returnAddressId: number;
    returnAddress?: Address;

    deliverymanId?: number;
    deliveryman?: User;
    deliverymanReturnId?: number;
    deliverymanReturn?: User;
    customerId: number;
    customer?: User;
    
    orderProducts: RealProduct[];
    totalPrice?: number;
}

File: API/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace API.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime OrderDate { get; set; }

        public DateTime RequiredDate { get; set; }
        public DateTime? ShippedDate { get; set; }

        public DateTime RequiredReturnDate { get; set; }
        public DateTime? ReturnDate { get; set; }

        public string Status { get; set; }

        public string Comments { get; set; }

        public int DeliverymanId { get; set; }
        public AppUser Deliveryman { get; set; }

        public int CustomerId { get; set; }
        public AppUser Customer { get; set; }

        public int ShippedAddressId { get; set; }
        public Address ShippedAddress { get; set; }

        public int ReturnAddressId { get; set; }
        public Address ReturnAddress { get; set; }

        public ICollection<OrderProduct> OrderProducts { get; set; }
    }
}